src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,409 @@
+    @Override
+    public Pair<List<SensitivityValue>, Map<String, List<SensitivityValue>>> analyse(Network network, List<SensitivityFactor> factors,
+                                                                                     List<Contingency> contingencies, LoadFlowParameters lfParameters,
+                                                                                     OpenLoadFlowParameters lfParametersExt) {
         Objects.requireNonNull(network);
         Objects.requireNonNull(factors);
         Objects.requireNonNull(lfParametersExt);
 
-        // create LF network (we only manage main connected component)
+        // create the network (we only manage main connected component)
         List<LfNetwork> lfNetworks = LfNetwork.load(network, lfParametersExt.getSlackBusSelector());
         LfNetwork lfNetwork = lfNetworks.get(0);
+        checkContingencies(lfNetwork, contingencies);
+        checkSensitivities(network, lfNetwork, factors);
+        LazyConnectivity connectivity = new LazyConnectivity(lfNetwork);
 
         // run DC load
-        Map<String, Double> functionReferenceByBranch = new HashMap<>();
-        DcLoadFlowParameters dcLoadFlowParameters = new DcLoadFlowParameters(lfParametersExt.getSlackBusSelector(), matrixFactory,
-                true, lfParametersExt.isDcUseTransformerRatio(), lfParameters.isDistributedSlack(),  lfParameters.getBalanceType());
-        DcLoadFlowResult dcLoadFlowResult = new DcLoadFlowEngine(lfNetworks, dcLoadFlowParameters).run();
-        for (LfBranch branch : dcLoadFlowResult.getNetwork().getBranches()) {
-            functionReferenceByBranch.put(branch.getId(), branch.getP1() * PerUnit.SB);
-        }
+        Map<String, Double> functionReferenceByBranch = getFunctionReferenceByBranch(lfNetworks, lfParameters, lfParametersExt);
 
         // create DC equation system for sensitivity analysis
         EquationSystem equationSystem = DcEquationSystem.create(lfNetwork, new VariableSet(),
                 new DcEquationSystemCreationParameters(false, true, true, lfParametersExt.isDcUseTransformerRatio()));
 
-        // index factors by variable configuration to compute minimal number of DC state
-        Map<SensitivityVariableConfiguration, SensitivityFactorGroup> factorsByVarConfig = indexFactorsByVariableConfig(network, factors, lfNetwork, lfParameters);
+        // index factors by variable configuration to compute minimal number of states
+        List<SensitivityFactorGroup> factorsGroups = createFactorGroups(network, factors);
+
+        if (factorsGroups.isEmpty()) {
+            return Pair.of(Collections.emptyList(), Collections.emptyMap());
+        }
 
-        if (factorsByVarConfig.isEmpty()) {
-            return Collections.emptyList();
+        // Compute the partipation for every factor
+        List<ParticipatingElement> participatingElements = null;
+        Function<String, Map<String, Double>> getParticipationForBus;
```

Comment:
On this particular line, it is not needed because connectivity is guaranteed, but when we process the contingencies, we need to know on what bus the injection is made, to know in which connected component we will take the elements participating to the load balancing